dropwizard-jdbi/src/main/java/io/dropwizard/jdbi/DBIHealthCheck.java

```diff
@@ -1,23 +1,40 @@
 package io.dropwizard.jdbi;
 
 import com.codahale.metrics.health.HealthCheck;
+import io.dropwizard.util.Duration;
 import org.skife.jdbi.v2.DBI;
 import org.skife.jdbi.v2.Handle;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
 
 public class DBIHealthCheck extends HealthCheck {
     private final DBI dbi;
     private final String validationQuery;
-
-    public DBIHealthCheck(DBI dbi, String validationQuery) {
```

Comment:
That's a good point. If we use those as defaults, it effectively means that health check won't be time bound. The callable will execute on the thread of the caller, and when it's finished the caller will retrieve the result with a timeout of zero; but it will always be ready. This seems like a good way to support backwards compatibility. Though the DBIFactory should pass in a executor.